MonoGame.Framework/Graphics/Texture2D.cs

```diff
@@ -52,6 +52,9 @@ protected Texture2D(GraphicsDevice graphicsDevice, int width, int height, bool m
             this.height = height;
             this._format = format;
             this._levelCount = mipmap ? CalculateMipLevels(width, height) : 1;
+
+            this._comparisonCode = _nextComparisonCode;
+            _nextComparisonCode++;

```

Comment:
Actually this would be better...

```
this._comparisonCode = Interlocked.Increment(ref _nextComparisonCode);
```

This not only handles overflow correctly, but it also makes this thread safe which is important.
```